### PR TITLE
Add Tron Stake v2

### PR DESCRIPTION
**Pull request by peachbits**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204436465323062

### CODE REVIEW

File: src/plugins/stake-plugins/types.ts [hunk 1]
```diff
@@ -160,8 +160,28 @@ export interface StakePosition {
 // Stake Plugin
 // -----------------------------------------------------------------------------
 
+export interface StakePolicyFilter {
+  wallet?: EdgeCurrencyWallet
+  currencyCode?: string
```
Comment (swansontec) [inline src/plugins/stake-plugins/types.ts:168]: I don't like seeing `currencyCode` used for any money stuff. It needs to become a purely a display thing going forward, not something we can compare based on.

However, I see that currencyCode exists in a bunch of places here, so maybe it's too hard to fix. If it really is a hard fix, I guess we can leave it.

File: src/plugins/stake-plugins/currency/tronStakePlugin.ts [hunk 2]
```diff
@@ -106,15 +146,15 @@ export const makeTronStakePlugin = async (): Promise<StakePlugin> => {
       const policy = getPolicyFromId(stakePolicyId)
       const resource = policy.rewardAssets[0].currencyCode
       const isStake = action === 'stake'
-
+      console.log('156. policy.deprecated', policy.deprecated)
```
Comment (swansontec) [inline src/plugins/stake-plugins/currency/tronStakePlugin.ts:156]: Leftover?

**Review by swansontec — APPROVED** [review]
Go ahead and delete the extra comment. If you decide to attack the currencyCode thing, then OK, I will do another review.

File: src/plugins/stake-plugins/types.ts (hunk 1 above)
Comment (peachbits) [reply src/plugins/stake-plugins/types.ts:168]: That's a good point. I looked at the potential work involved and currencyCode is pretty pervasive throughout the staking plugin so I'd rather punt

**Pull request merged**